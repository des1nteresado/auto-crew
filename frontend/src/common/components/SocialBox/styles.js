import instagram from 'static/images/instagram.svg';
import instagramActive from 'static/images/instagram-hover.svg';
import facebook from 'static/images/facebook.svg';
import facebookActive from 'static/images/facebook-hover.svg';
import pinterest from 'static/images/pinterest.svg';
import pinterestActive from 'static/images/pinterest-hover.svg';
import twitter from 'static/images/twitter.svg';
import twitterActive from 'static/images/twitter-hover.svg';

const styles = (theme) => ({
  socialBox: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    maxWidth: 250,
    [theme.breakpoints.down('xs')]: {
      margin: 'auto',
      marginTop: 30,
    },
  },
  instagram: {
    cursor: 'pointer',
    width: 16,
    height: 16,
    backgroundImage: `url(${instagram})`,
    '&:hover': {
      backgroundImage: `url(${instagramActive})`,
    },
    '&:active': {
      backgroundImage: `url(${instagramActive})`,
    },
  },
  facebook: {
    cursor: 'pointer',
    width: 8,
    height: 16,
    backgroundImage: `url(${facebook})`,
    '&:hover': {
      backgroundImage: `url(${facebookActive})`,
    },
    '&:active': {
      backgroundImage: `url(${facebookActive})`,
    },
  },
  pinterest: {
    cursor: 'pointer',
    width: 17,
    height: 16,
    backgroundImage: `url(${pinterest})`,
    '&:hover': {
      backgroundImage: `url(${pinterestActive})`,
    },
    '&:active': {
      backgroundImage: `url(${pinterestActive})`,
    },
  },
  twitter: {
    cursor: 'pointer',
    width: 21,
    height: 16,
    backgroundImage: `url(${twitter})`,
    '&:hover': {
      backgroundImage: `url(${twitterActive})`,
    },
    '&:active': {
      backgroundImage: `url(${twitterActive})`,
    },
  },
});

export default styles;
