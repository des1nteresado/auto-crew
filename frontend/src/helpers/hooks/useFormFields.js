import { useState } from 'react';

export const useFormFields = (initialState) => {
  const [fields, setValue] = useState(initialState);

  return [
    fields,
    (event) => {
      if (event.target.type === 'checkbox') {
        setValue({
          ...fields,
          [event.target.name]: event.target.checked,
        });
      } else {
        setValue({
          ...fields,
          [event.target.name]: event.target.value,
        });
      }
    },
  ];
};
