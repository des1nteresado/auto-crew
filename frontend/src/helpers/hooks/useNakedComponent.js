export const useNakedComponent = (component) => {
  return component?.type?.Naked;
};
