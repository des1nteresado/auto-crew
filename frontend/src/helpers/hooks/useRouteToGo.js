import { useCallback } from 'react';
import { useHistory } from 'react-router-dom';

export const useRouteToGo = (path) => {
  const history = useHistory();

  return useCallback(() => history.push(path), [path, history]);
};
