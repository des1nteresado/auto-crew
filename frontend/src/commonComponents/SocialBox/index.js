import React from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { Box, Link, withStyles } from '@material-ui/core';

import styles from './styles';

const SocialBox = ({ classes, onClick, style, socialNetworks }) => {
  return (
    <Box className={clsx(classes.socialBox, style && style)}>
      {socialNetworks &&
        socialNetworks.map((network) => (
          <Link key={network.name} href={network.link}>
            <Box onClick={onClick} className={classes[`${network.name}`]} />
          </Link>
        ))}
    </Box>
  );
};

SocialBox.defaultProps = {
  style: '',
  socialNetworks: [],
  onClick: () => {},
};

SocialBox.propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  socialNetworks: PropTypes.oneOfType([PropTypes.array]),
  style: PropTypes.string,
  onClick: PropTypes.func,
};

export default React.memo(withStyles(styles)(SocialBox));
