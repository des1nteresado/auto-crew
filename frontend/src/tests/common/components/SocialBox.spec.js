import React from 'react';
import { shallow } from 'enzyme';

import SocialBox from 'common/components/SocialBox';
import { useNakedComponent } from 'helpers/hooks/useNakedComponent';

const defaultProps = {
  socialNetworks: [
    {
      name: 'name1',
      link: 'link1',
    },
    {
      name: 'name2',
      link: 'link2',
    },
  ],
  style: 'style',
};

const setUp = (props) => shallow(<SocialBox {...defaultProps} {...props} />).dive();

describe('SocialBox', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it('renders SocialBox component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders correct number of social links', () => {
    const socialNetworks = component.find('WithStyles(ForwardRef(Link))');

    expect(socialNetworks).toHaveLength(defaultProps.socialNetworks.length);
  });

  it('renders SocialBox component with provided styles correctly', () => {
    const box = component.find('Styled(MuiBox)').first();
    expect(box.props().className.includes(defaultProps.style)).toBeTruthy();
  });

  describe('default props', () => {
    it('should use default onInputChange', () => {
      const nakedComponent = useNakedComponent(SocialBox);

      const result = nakedComponent.defaultProps.onClick();
      expect(result).toBe(undefined);
    });
  });
});
