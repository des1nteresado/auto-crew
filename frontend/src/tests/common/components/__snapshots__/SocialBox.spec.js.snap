// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialBox renders SocialBox component correctly 1`] = `
<Styled(MuiBox)
  className="SocialBox-socialBox-1 style"
>
  <WithStyles(ForwardRef(Link))
    href="link1"
    key="name1"
  >
    <Styled(MuiBox)
      onClick={[Function]}
    />
  </WithStyles(ForwardRef(Link))>
  <WithStyles(ForwardRef(Link))
    href="link2"
    key="name2"
  >
    <Styled(MuiBox)
      onClick={[Function]}
    />
  </WithStyles(ForwardRef(Link))>
</Styled(MuiBox)>
`;
